addons/implied-repos/common/src/main/java/org/commonjava/indy/implrepo/change/ImpliedRepositoryDetector.java

```diff
@@ -399,19 +400,31 @@ else if ( !config.isIncludeSnapshotRepos() && !repo.isReleasesEnabled() )
                             e );
                 }
 
+                final RemoteRepository ref = creator.createFrom( gav, repo, LoggerFactory.getLogger( creator.getClass() ) );
```

Comment:
What's the purpose of this variable? It looks like we're cloning it below, if there are no matching remote repositories (by URL). Is this just to get the creation check done up front?

Reply:
That looks so, because we use this variable in several places, except the matching part, but also the following if section and else section. Just want to abstract it here not create it several times.